game-core/src/main/java/games/strategy/engine/framework/startup/ui/pbem/test/post/TestPostAction.java

```diff
@@ -0,0 +1,76 @@
+package games.strategy.engine.framework.startup.ui.pbem.test.post;
+
+import com.google.common.base.Preconditions;
+import games.strategy.engine.ClientContext;
+import games.strategy.engine.posted.game.pbf.NodeBbForumPoster;
+import java.awt.Graphics;
+import java.awt.image.BufferedImage;
+import java.io.File;
+import java.io.IOException;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.ExecutionException;
+import java.util.function.BiConsumer;
+import java.util.function.Supplier;
+import java.util.logging.Level;
+import javax.imageio.ImageIO;
+import lombok.AllArgsConstructor;
+import lombok.extern.java.Log;
+import org.triplea.java.TimeManager;
+
+@Log
+@AllArgsConstructor
+public class TestPostAction implements BiConsumer<String, Integer> {
+
+  private final Supplier<TestPostProgressDisplay> testPostProgressDisplayFactory;
+
+  @Override
+  public void accept(final String forumName, final Integer topicId) {
+    Preconditions.checkNotNull(forumName);
+    Preconditions.checkArgument(topicId > 0);
+
+    final TestPostProgressDisplay testPostProgressDisplay = testPostProgressDisplayFactory.get();
+
+    new Thread(
+            () -> {
+              File f = null;
```

Comment:
All logic in this method is extracted from the view. Any interactions with swing are delegated to `TestPostProgressDisplay`